### PR TITLE
Np 1300 more search parameters

### PR DESCRIPTION
**Pull request by svenngjeto**
'Results' query parameter is send to elasticsearch, if not present a default value is used. Parameter 'search_after' is removed for now.

### CODE REVIEW

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 1]
```diff
@@ -53,11 +56,14 @@ public void defaultConstructorThrowsIllegalStateExceptionWhenEnvironmentNotDefin
...
-    public void processInputReturnsNullWhenInputIsEmpty() throws ApiGatewayException {
+    public void processThrowsExceptionWhenInputIsEmpty() throws ApiGatewayException {
         SearchResourcesRequest input = mock(SearchResourcesRequest.class);
         RequestInfo requestInfo = mock(RequestInfo.class);
+        doThrow(IllegalArgumentException.class).when(requestInfo).getQueryParameter(any());
         Context context = mock(Context.class);
-        assertThrows(InputException.class, () ->  searchResourcesApiHandler.processInput(input, requestInfo, context));
+        assertThrows(ApiGatewayException.class, () ->  searchResourcesApiHandler.processInput(input,
+                requestInfo,
+                context));
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:66]: Line 80, replace with a call to:

```
    @SuppressWarnings("unchecked")
    private HttpResponse<String> getHttpResponseMock() {
        return mock(HttpResponse.class);
    }
```

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: search-resources-api/src/main/java/no/unit/nva/search/Constants.java [hunk 2]
```diff
@@ -1,12 +1,6 @@
 package no.unit.nva.search;
 
 public class Constants {
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/Constants.java:3]: Class is gone..

File: search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchRestClient.java [hunk 3]
```diff
@@ -103,12 +108,14 @@ private HttpRequest buildHttpRequest(String term) {
...
-                elasticSearchEndpointScheme, elasticSearchEndpointAddress,
-                elasticSearchEndpointIndex, term);
-        logger.debug("uriString={}",uriString);
-        return URI.create(uriString);
+    private URI createSearchURI(String term, String results) {
+        return  new URIBuilderTiny()
+                .setScheme(elasticSearchEndpointScheme)
+                .setHost(elasticSearchEndpointAddress)
+                .setPaths(elasticSearchEndpointIndex, ELASTIC_SEARCH_OPERATION)
+                .addQueryParameter(ELASTIC_QUERY_PARAMETER,term)
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchRestClient.java:116]: Reformatted and inlined

File: search-resources-api/src/main/java/no/unit/nva/search/RequestUtil.java [hunk 4]
```diff
@@ -23,15 +19,12 @@
      * @throws ApiGatewayException exception containing explanatory message when parameter missing or inaccessible
      */
     public static String getSearchTerm(RequestInfo requestInfo) throws ApiGatewayException {
+        return requestInfo.getQueryParameter(SEARCH_TERM_KEY);
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/RequestUtil.java:22]: Ghost exception removed

File: search-resources-api/src/test/java/no/unit/nva/search/ElasticSearchRestClientTest.java [hunk 5]
```diff
@@ -8,6 +8,9 @@
 import java.io.IOException;
 import java.net.http.HttpClient;
 
+import static no.unit.nva.search.ElasticSearchRestClient.ELASTICSEARCH_ENDPOINT_ADDRESS_KEY;
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/ElasticSearchRestClientTest.java:11]: Fixed

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 1 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:66]: Code moved

File: search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchRestClient.java [hunk 6]
```diff
@@ -36,6 +36,7 @@
     public static final String HITS_JSON_POINTER = "/hits/hits";
     public static final String ERROR_READING_RESPONSE_FROM_ELASTIC_SEARCH =
             "Error when reading response from ElasticSearch";
+    public static final String RESULTSIZE_PATTERN = "&size=&s";
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchRestClient.java]: Removed unused constants again

**Review by brinxmat — APPROVED** [review]

**Pull request merged**